doc/fgpu.design.rst

```diff
@@ -177,51 +177,129 @@ slow and require multiple passes over the memory, because
 
 Real-to-complex transform
 ~~~~~~~~~~~~~~~~~~~~~~~~~
-Let's introduce some notation to see how this works. Let :math:`n` be the
-number of channels, :math:`x_i` be the (real) time-domain samples and
-:math:`X_k` be the Fourier transform (we divide by :math:`2n` rather than
-:math:`n` because that's the number of input samples used).
+Now for some notation to see how this works. Let :math:`z` be the array of samples (:math:`x_{i}`) of
+length :math:`N`. For clarity, array :math:`x` has length :math:`2N` but is divided into even- and odd- samples
+each of length :math:`N`. In other words, each adjacent pair of real values in :math:`x` is interpreted as the real
+and imaginary components of a complex value.
+
+Let :math:`u_i = x_{2i}` and :math:`v_i = x_{2i+1}`, and let :math:`z_i = u_i + jv_i = x_{2i} + j x_{2i+1}`.
+We will start by computing the Fourier transform of :math:`z`. This is convenient to do because :math:`z` has the
+identical memory layout to :math:`x`, so we can obtain it just by viewing the same memory as containing N complex
+numbers instead of 2N real numbers. In other words, each adjacent pair of real values in :math:`x` is interpreted
+as the real and imaginary components of a complex value.
+
+Let :math:`U`, :math:`V` and :math:`Z` denote the Fourier transforms of :math:`u`, :math:`v` and :math:`z` respectively.
+Since the Fourier transform is a linear operator and we defined :math:`z = u + jv`, we also have :math:`Z = U + jV`.
+
+It is important to remember that both :math:`u` and :math:`v` are real-valued, so :math:`U`
+and :math:`V` are Hermitian symmetric. By re-arranging things we can reconstruct :math:`U` and
+:math:`V` from :math:`Z` using Hermitian symmetry properties. Let :math:`U'`
+be :math:`U` with reversed indices i.e., :math:`U'_k = U_{-k}` where indices are taken
+modulo :math:`n`.
+
+Hermitian symmetry means that :math:`U'_k = U_{-k} = \overline{U_k}` where
+the 'overline' in :math:`\overline{U_k}` denotes conjugation. This is effectively saying that by taking the
+reverse indices in :math:`U_k` we get a conjugated result.
+
+Why is this so? Going back to the original definition for the DFT we saw the complex exponential
+:math:`e^{\frac{-2\pi j}{2N}\cdot ik}` has a variable :math:`k` where :math:`k` represents the
+frequency component under computation for the input sequence :math:`x_i`. If :math:`k` is reversed
+(i.e. negative) the complex exponential changes to :math:`e^{\frac{2\pi j}{2n}\cdot i(k)}` as the negaitive
+in :math:`-k` multiplies out.
+
+Moving back to :math:`U` and :math:`V` components then :math:`U' = \overline{U}` and similarly :math:`V' = \overline{V}`.
+Going full circle (for completeness) if one was to conjugate the reversal (i.e. :math:`\overline{U'}`) then since
+:math:`U' = \overline{U}`, :math:`\overline{U'} = \overline{\overline{U}} = U`. Likewise for
+:math:`\overline{V'} = \overline{\overline{V}} = V`.
```

Comment:
@james-smith-za again I think we need a tie-breaker from you - I think stating that conjugating twice cancels out doesn't really add anything.